src/renderer/content.ts

```diff
@@ -1,4 +1,93 @@
-import { EditorId } from '../interfaces';
+import { USER_DATA_PATH } from './constants';
+import { EditorId, EditorValues } from '../interfaces';
+import { fancyImport } from '../utils/import';
+import { readFiddle } from '../utils/read-fiddle';
+
+import * as fetchType from 'node-fetch';
+import * as fsType from 'fs-extra';
+import * as path from 'path';
+import * as semver from 'semver';
+
+const decompress = require('decompress');
+
+// parent directory of all the downloaded template fiddles
+const TEMPLATES_DIR = path.join(USER_DATA_PATH, 'Templates');
+
+// location of the fallback template fiddle used iff downloading failed
+const STATIC_TEMPLATE_DIR = path.resolve(
+  __dirname,
+  '../../static/electron-quick-start',
+);
+
+/**
+ * Ensure we have a fiddle for the specified Electron branch.
+ * If we don't have it already, download it from electron-quick-start.
+ *
+ * @param {branch} Electron branchname, e.g. `12-x-y` or `master`
+ * @returns {Promise<string>} Path to the folder where the fiddle is kept
+ */
+async function prepareTemplate(branch: string): Promise<string> {
+  let folder = path.join(TEMPLATES_DIR, `electron-quick-start-${branch}`);
+
+  try {
+    // if we don't have it, download it
+    const fs = await fancyImport<typeof fsType>('fs-extra');
+    if (!fs.existsSync(folder)) {
+      console.log(`Content: Downloading template for ${branch}`);
+      const url = `https://github.com/electron/electron-quick-start/archive/${branch}.zip`;
+      const fetch = (await fancyImport<any>('node-fetch')).default;
+      const response: fetchType.Response = await fetch(url);
+      const buffer: Buffer = await response.buffer();
+
+      console.log(`Content: Unzipping template for ${branch}`);
+      await fs.ensureDir(TEMPLATES_DIR);
+      await decompress(buffer, TEMPLATES_DIR);
```

Comment:
It seems like this code fizzles out when given a non-existent major version, leading to the editors being blank.

![image](https://user-images.githubusercontent.com/16010076/110842456-f50f7d80-825b-11eb-8dda-a1f9db39f971.png)

I debugged a bit here, and when `buffer` value contains a `404 not found` response error (e.g. when hitting https://github.com/electron/electron-quick-start/archive/867-x-y.zip) , it seems like `decompress` will take that buffer and just silently fail.

Reply:
Ahh yes, good catch! I should've thought of that use case.

When we 404, let's fall back to `master`

Reply:
I've updated this in 32f690d0c4f6f1547acb1800771a1d81a0487166 to check fetch's response's OK status. I didn't realize it wouldn't reject on 404/500.

This still leaves a bit of a smell IMO, that we're pinging GitHub every time we use a nonexistent branch. I've added a9cdbe3a3aaf00ee3fc5e5e994879f219e25c94c s.t. we only ping GitHub for known remote release branches to solve this.